tests/test_indexing.py

```diff
@@ -19,13 +19,58 @@
 
 """Tests indexing of records."""
 
+import time
+
 from invenio.testsuite import InvenioTestCase
 
 
 class IndexingTests(InvenioTestCase):
+
     def test_reindexing(self):
         """Test that all records are indexed"""
         from invenio_search.api import Query
         # Note that the number might change if new demo records are
         # introduced.
         self.assertEqual(len(Query("").search()), 426)
+
+    def test_recids(self):
+        """Test that recids API works"""
+        from invenio_search.api import Query
+        # Note that the number might change if new demo records are
+        # introduced.
+        self.assertEqual(len(Query("").search().recids), 426)
+
+    def test_recids_collection(self):
+        """Test that recids API works"""
+        from invenio_search.api import Query
+        # Note that the number might change if new demo records are
+        # introduced.
+        self.assertEqual(len(Query("").search(collection='HEP').recids), 426)
+
+    def test_recids_restricted_collection(self):
+        """Test that recids API works"""
+        from invenio_ext.sqlalchemy import db
+        from invenio_search.api import Query
+        from invenio_records.api import Record
+
+        self.assertEqual(len(Query("").search(collection='CDF Internal Notes').recids), 0)
+
+        test_recid = 1396160
+        original_record = dict(Record.get_record(test_recid))
+
+        rec = Record.get_record(test_recid)
+        rec['collections'].append({'primary': 'CDF-INTERNAL-NOTE'})
+        rec.commit()
+
+        # Wait for indexing
+        time.sleep(10)
+
+        self.assertEqual(len(Query("").search(collection='CDF Internal Notes').recids), 0)
+        self.assertEqual(len(Query("").search(
+            collection='CDF Internal Notes',
+            user_info={'precached_permitted_restricted_collections': ['CDF Internal Notes']}
+        ).recids), 1)
+
+        rec = Record.get_record(test_recid)
+        rec.model.json = original_record
+        db.session.commit()
```

Comment:
Better put this last part in the`finally` of a  `try..finally`, so that we are sure it's always executed also in case of errors. Otherwise the test would live traces behind.